### PR TITLE
Reverse binary

### PR DESCRIPTION
**Pull request by ysard**
add reverse complementary algo with bitwise operations.
improvements in run.py and Makefile

### CODE REVIEW

**Comment by natir** [comment]
Thank for your pull request but, please i need you make some modification before integrate your pull request :
- Please remove your Makefile isn't required 
- In CMakeList use -O2 and not -Ofast (some problem with [floating-point operation](https://www.quora.com/When-should-you-use-the-different-GCC-optimization-flags-e-g-O2))
- Remove your modification in generate_data.py and in run.py (i won't output become a default value and remove some comment code in run.py)
- Add in cpp/algorithm.md some explain of your algorithm

File: cpp/inc/rev_comp_bin.hpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#include "arev_comp.hpp"
+
+///////////////////////////////////////////////////////////////////////////////
+/// \brief The rev_comp_bin class
+///
+class rev_comp_bin : public arev_comp
+{
+public:
+    rev_comp_bin();
+    virtual std::string run(std::string sequence);
```
Comment (natir) [inline cpp/inc/rev_comp_bin.hpp]: virtual isn't required

File: cpp/inc/rev_comp_bin.hpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+private:
+    // Initialises the mapping between ASCII & Binary data
+    std::array<uint8_t, 85> m_mappingDataADN;
+};
+
+///////////////////////////////////////////////////////////////////////////////
+/// \brief The LifeGenerator class
+///
+class LifeGenerator
```
Comment (natir) [inline cpp/inc/rev_comp_bin.hpp]: This class isn't required

File: cpp/src/rev_comp_bin.cpp [hunk 3]
```diff
@@ -0,0 +1,210 @@
+#include "rev_comp_bin.hpp"
+
+// Yes i fuck namespaces in cpp !
+using namespace std;
+
+///////////////////////////////////////////////////////////////////////////////
+
+LifeGenerator::LifeGenerator():
```
Comment (natir) [inline cpp/src/rev_comp_bin.cpp]: Remove this please

**Comment by natir** [comment]
Oups i see some other change
- Remove unused code (LifeGenerator)
- Translate or remove your comment in french
- Remove debug instruction and commented debug instruction

**Comment by ysard** [comment]
i did not see the first notifications.
I will redo my PR.

In my opinion, unless there is a good documentation; Makefile is absolutely necessary to test the code in less than 1h after git clone...

Same things for run.py and generate_data.py. We don't have to guess arguments.
A good exemple is --list-lang parameter. Currently, there is no other language and yet this option is not optional.

**Pull request closed without merging**